utils/interfaces/_library/miscs.py

```diff
@@ -32,3 +32,19 @@ def __call__(self, span):
                     raise ValueError(f'{tagKey} tag in span\'s meta should be "{expectValue}", not "{actualValue}"')
 
         return True
+
+
+class _NotSpanTagValidator:
```

Comment:
This is the legacy way of doing validation, I'll adapt it with the new way once everything is ok.

Reply:
What is the new way? I was just following the `_SpanTagValidator`

Reply:
it's using directly the span accessors inside you test methd: 

```python
def test_tags_absence(self):
  forbidden_tags = ("tag_1", "tag_2")

  for data, trace, span in interfaces.library.get_spans(request=request):
     for tag in forbidden_tags:
        assert tag not in span["meta"]
```

Having all the test logic in test method makes the test easier to understand on debug. 

Two rules of thumb : 

* data must be present and clear in test method, or at least, very close to it
* non-python engineer must easily understand the test code by reading it

Reply:
So does that mean I should replace all calls to `interfaces.library.add_span_tag_validation` with code like the above as well?

Reply:
I missed your comment, sorry. So the answer is yes. I need to remove the legacy way, my bad.